File: output/CodeLlama/avatar/Java/Python/atcoder_AGC021_B.py
import math
import sys

N = int(sys.stdin.readline().rstrip())
x = [0] * N
y = [0] * N
for i in range(N):
    x[i], y[i] = map(int, sys.stdin.readline().split())

for i in range(N):
    thetas = []
    for j in range(N):
        if i == j:
            continue
        thetas.append(math.atan2(y[j] - y[i], x[j] - x[i]))
    thetas.sort()
    thetas.append(thetas[0] + 2 * math.pi)
    ans = 0
    for k in range(N - 1):
        ans = max(ans, thetas[k + 1] - thetas[k] - math.pi)
    print(ans / (math.pi * 2))

File: output/CodeGen/avatar/Python/Java_handraft/atcoder_AGC021_B.java
public class atcoder_AGC021_B {
    public double findRadius ( int[] x, int[] y ) {
        int N = x.length;
        int[] X = new int[N];
        int[] Y = new int[N];
        for ( int i = 0 ; i < N ; i ++ ) {
            X[i] = x[i];
            Y[i] = y[i];
        }
        Arrays.sort(X);
        Arrays.sort(Y);
        double ans = 0;
        for ( int i = 0 ; i < N ; i ++ ) {
            double[] D = new double[N];
            for ( int j = 0 ; j < N ; j ++ ) {
                if ( j!= i ) {
                    D[j] = Math.atan2(X[i] - X[j], Y[i] - Y[j]);
                }
            }
            D[i] += 2 * Math.PI;
            for ( int j = 0 ; j < N ; j ++ ) {
                if ( j!= i ) {
                    if ( D[j] - D[i] >= Math.PI ) {
                        ans += ( D[j] - D[i] ) - Math.PI;
                    }
                }
            }
        }
        return ans / ( 2 * Math.PI );
    }
}